src/net_builtin.c

```diff
@@ -476,14 +476,22 @@ static void rx_dhcp_server(struct mg_tcpip_if *ifp, struct pkt *pkt) {
         53, 1, msg,                 // Message type
         1,  4, 0,   0,   0,   0,    // Subnet mask
         54, 4, 0,   0,   0,   0,    // Server ID
-        12, 3, 'm', 'i', 'p',       // Host name: "mip"
         51, 4, 255, 255, 255, 255,  // Lease time
-        255                         // End of options
     };
+    // DHCP hostname: guarantee to limit its length by sizeof(ifp->dhcp_name)
+    size_t name_len = strlen(ifp->dhcp_name) % sizeof(ifp->dhcp_name);
```

Comment:
there's room for 12 chars. if `ifp->dhcp_name[11] != 0` then `%` will return `0`. It can't be larger anyway, no room.
```
ifp->dhcp_name[sizeof(ifp->dhcp_name) - 1] = `\0`;
```
That's a guarantee

Reply:
Yes! Done that way